src/drivers/actuators/vertiq_io/vertiq_client_manager.cpp

```diff
@@ -0,0 +1,208 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#include "vertiq_client_manager.hpp"
+
+VertiqClientManager::VertiqClientManager(VertiqSerialInterface *serial_interface) :
+	_serial_interface(serial_interface),
+	_broadcast_prop_motor_control(_kBroadcastID), //Initialize with a module ID of 63 for broadcasting
+	_broadcast_arming_handler(_kBroadcastID)
+{
+	_client_array[0] = &_broadcast_prop_motor_control;
+	_client_array[1] = &_broadcast_arming_handler;
+}
+
+void VertiqClientManager::Init(uint8_t object_id)
+{
+	InitVertiqClients(object_id);
+	InitComboEntries();
+
+	_object_id_now = object_id;
+}
+
+void VertiqClientManager::InitVertiqClients(uint8_t object_id)
+{
+	_prop_input_parser_client = new EscPropellerInputParserClient(object_id);//&prop_input_parser;
+	_client_array[_clients_in_use] = _prop_input_parser_client;
+	_clients_in_use++;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	_ifci_client = new IQUartFlightControllerInterfaceClient(object_id);
+	_client_array[_clients_in_use] = _ifci_client;
+	_clients_in_use++;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	_voltage_superposition_client = new VoltageSuperPositionClient(object_id);
+	_client_array[_clients_in_use] = _voltage_superposition_client;
+	_clients_in_use++;
+
+	_pulsing_rectangular_input_parser_client = new PulsingRectangularInputParserClient(object_id);
+	_client_array[_clients_in_use] = _pulsing_rectangular_input_parser_client;
+	_clients_in_use++;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+	//We're done with determining how many clients we have, let the serial interface know
+	_serial_interface->SetNumberOfClients(_clients_in_use);
+}
+void VertiqClientManager::InitComboEntries()
```

Comment:
go through and initialize all of the entry wrappers. they get a PX4 parameter (passed with the param_find) and an IQUART entry